### ISSUE TITLE
Compilation problem

### ISSUE THREAD

**Issue by wwood** [10XGenomics/rust-pseudoaligner#7]
Hi,

Thanks for writing this code, I am finding it useful.

However, upon running `cargo update` recently it started failing:

```
$ cargo update
    Updating crates.io index
    Updating git repository `https://github.com/10XGenomics/rust-boomphf`
    Updating git repository `https://github.com/10XGenomics/rust-debruijn`
    Updating git repository `https://github.com/k3yavi/pretty-env-logger`
error: failed to select a version for `boomphf`.
    ... required by package `debruijn_mapping v0.2.0 (/home/ben/git/rust-tmp/rust-pseudoaligner)`
versions that meet the requirements `*` are: 0.5.2

the package `debruijn_mapping` depends on `boomphf`, with features: `fast-constructors` but `boomphf` does not have these features.


failed to select a version for `boomphf` which could resolve this conflict
```

Removing the "fast-constructors" lets cargo update work, but then I hit other compilation errors:

```
$ cargo build
   Compiling debruijn_mapping v0.2.0 (/home/ben/git/rust-tmp/rust-pseudoaligner)
error[E0308]: mismatched types
   --> src/build_index.rs:125:63
    |
125 |       let (phf, _): (BoomHashMap2<K, Exts, EqClassIdType>, _) = filter_kmers(
    |  _______________________________________________________________^
126 | |         &filter_input,
127 | |         summarizer,
128 | |         STRANDED,
129 | |         REPORT_ALL_KMER,
130 | |         MEM_SIZE,
131 | |     );
    | |_____^ expected struct `boomphf::hashmap::BoomHashMap2`, found a different struct `boomphf::hashmap::BoomHashMap2`
    |
    = note: expected type `(boomphf::hashmap::BoomHashMap2<K, debruijn::Exts, u32>, _)`
               found type `(boomphf::hashmap::BoomHashMap2<_, debruijn::Exts, u32>, std::vec::Vec<_>)`
note: Perhaps two different versions of crate `boomphf` are being used?
   --> src/build_index.rs:125:63
    |
125 |       let (phf, _): (BoomHashMap2<K, Exts, EqClassIdType>, _) = filter_kmers(
    |  _______________________________________________________________^
126 | |         &filter_input,
127 | |         summarizer,
128 | |         STRANDED,
129 | |         REPORT_ALL_KMER,
130 | |         MEM_SIZE,
131 | |     );
    | |_____^

error[E0308]: mismatched types
   --> src/build_index.rs:133:40
    |
133 |     compress_kmers_with_hash(STRANDED, ScmapCompress::new(), &phf)
    |                                        ^^^^^^^^^^^^^^^^^^^^
    |                                        |
    |                                        expected reference, found struct `debruijn::compression::ScmapCompress`
    |                                        help: consider borrowing here: `&ScmapCompress::new()`
    |
    = note: expected type `&_`
               found type `debruijn::compression::ScmapCompress<_>`

error[E0308]: mismatched types
   --> src/build_index.rs:133:62
    |
133 |     compress_kmers_with_hash(STRANDED, ScmapCompress::new(), &phf)
    |                                                              ^^^^ expected struct `boomphf::hashmap::BoomHashMap2`, found a different struct `boomphf::hashmap::BoomHashMap2`
    |
    = note: expected type `&boomphf::hashmap::BoomHashMap2<K, debruijn::Exts, u32>` (struct `boomphf::hashmap::BoomHashMap2`)
               found type `&boomphf::hashmap::BoomHashMap2<K, debruijn::Exts, u32>` (struct `boomphf::hashmap::BoomHashMap2`)
note: Perhaps two different versions of crate `boomphf` are being used?
   --> src/build_index.rs:133:62
    |
133 |     compress_kmers_with_hash(STRANDED, ScmapCompress::new(), &phf)
    |                                                              ^^^^

error[E0308]: mismatched types
   --> src/build_index.rs:140:30
    |
140 |     compress_graph(STRANDED, ScmapCompress::new(), combined_graph, None)
    |                              ^^^^^^^^^^^^^^^^^^^^
    |                              |
    |                              expected reference, found struct `debruijn::compression::ScmapCompress`
    |                              help: consider borrowing here: `&ScmapCompress::new()`
    |
    = note: expected type `&_`
               found type `debruijn::compression::ScmapCompress<_>`

error[E0599]: no function or associated item named `new_parallel_with_keys` found for type `boomphf::Mphf<_>` in the current scope
   --> src/build_index.rs:155:31
    |
155 |     let mphf = boomphf::Mphf::new_parallel_with_keys(1.7, dbg, None, total_kmers, MAX_WORKER);
    |                               ^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `boomphf::Mphf<_>`

error: aborting due to 5 previous errors

Some errors occurred: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
error: Could not compile `debruijn_mapping`.

To learn more, run the command again with --verbose.
```

Are these easy to fix? Thanks.

**Comment by k3yavi** [comment]
just to give you heads up, I can compile a fresh version on my Ubuntu machine. However, I am not sure how can we make a version update for an old version w/ `fast-constructors`. I am guessing some cached file is messing up the compilation ?

**Comment by wwood** [comment]
Right, clone and compile works for me too. I think that's because the Cargo.lock is in the repo. Cargo update fails though with the above error.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by wwood** [comment]
It seems this was fixed in #6 and now merged - thanks.
